react/features/video-menu/components/AbstractDemoteButton.ts

```diff
@@ -0,0 +1,34 @@
+import { openDialog } from '../../base/dialog/actions';
+import { IconUsers } from '../../base/icons/svg';
+import AbstractButton, { IProps as AbstractButtonProps } from '../../base/toolbox/components/AbstractButton';
+
+import { DemoteToVisitorDialog } from './';
+
+export interface IProps extends AbstractButtonProps {
+
+    /**
+     * The ID of the participant that this button is supposed to kick.
+     */
+    participantID: string;
+}
+
+/**
+ * An abstract remote video menu button which demotes the remote participant.
+ */
+export default class AbstractDemoteButton extends AbstractButton<IProps> {
```

Comment:
I think we use this only on mobile. Is this correct?

If this is the case why do we need Abstract class at all?